autoload/neomake/makers/ft/d.vim

```diff
@@ -14,52 +14,66 @@ function! neomake#makers#ft#d#EnabledMakers()
     return l:makers
 endfunction
 
+function! s:findDubRoot()
+    "Look upwards for a dub.json or dub.sdl to find the root
+    "I did it like this because it's the only cross platform way I know of
+    let l:tmp_file = findfile("dub.json", ".;")
+    if(empty(l:tmp_file))
+        let l:tmp_file = findfile("dub.sdl", ".;")
+    endif
+    return l:tmp_file
+endfunction
+
 function! s:UpdateDub()
     "Add dub directories
-    let l:tmp_file = findfile("dub.json", ".;")
-    "dmd happily accepts an empty import string.
-    let l:dub_incs = "-I"
+    let l:tmp_file = s:findDubRoot()
     if neomake#utils#Exists('dub') && !empty(l:tmp_file)
         let l:tmp_dir = fnamemodify(l:tmp_file,':p:h')
-        "Get the dub dependencies from the dub describe command.
-        "vim doesn't seem to play nice with newlines
-        let l:dub = eval(substitute(system("dub describe --annotate --root=" . tmp_dir)
-                    \ ,'\v[\n\r]','','g'))
-        let l:tmp_arr = []
-        "Extract the packages from the dictionary
-        for l:package in l:dub.packages
-            let l:path = l:package.path
-            for l:importPath in l:package.importPaths
-                call add(tmp_arr, l:path . l:importPath)
-            endfor
-        endfor
-        let l:dub_incs .= join(tmp_arr, ":")
+        let l:dubCmd = "dub describe --data=import-paths --annotate \
+                    \--skip-registry=all --vquiet --data-list --root="
+        let l:output = system(l:dubCmd . tmp_dir)
+        if(v:shell_error == 0 && !empty(l:output))
+            " Is \n portable?
+            let s:dubImports = split(l:output, '\n')
+            call map(s:dubImports, '"-I" . v:val')
+            return
+        endif
     endif
-    return l:dub_incs
+    let s:dubImports = []
```

Comment:
This reads a bit backwards. I think this should go at the top of the function, and just set it again if dub is found. Then you also don't need to return early above.